plugin/yt_dlp_plugins/extractor/getpot_bgutil.py

```diff
@@ -27,22 +29,35 @@ def _get_pot(self, client: str, ydl: YoutubeDL, visitor_data=None, data_sync_id=
             po_token = self._get_pot_via_http(ydl, client, visitor_data, data_sync_id)
 
         return po_token
-    
-    def _get_pot_via_http(self, ydl, client, visitor_data, data_sync_id):
-        response = ydl.urlopen(Request('http://127.0.0.1:4416/get_pot', data=json.dumps({
-            'client': client,
-            'visitor_data': visitor_data,
-            'data_sync_id': data_sync_id
-        }).encode(), headers = {'Content-Type': 'application/json'}))
-
-        response_json = json.loads(response.read().decode('utf-8'))
 
+    def _get_pot_via_http(self, ydl, client, visitor_data, data_sync_id):
+        try:
+            response = ydl.urlopen(Request('http://127.0.0.1:4416/get_pot', data=json.dumps({
+                'client': client,
+                'visitor_data': visitor_data,
+                'data_sync_id': data_sync_id
+            }).encode(), headers = {'Content-Type': 'application/json'}))
+        except Exception as e:
+            raise RequestError(f"Error reaching POST /get_pot: {str(e)}")
+            
+        try:
+            response_json = json.loads(response.read().decode('utf-8'))
+        except Exception as e:
+            raise RequestError(f"Error parsing response JSON: {str(e)}. response = {response.read().decode('utf-8')}")
```

Comment:
why not always send a json and put the error message into a field like `"error"`/`"errors"`. the failure to parse the response JSON isn't expected.